internal/olm/operator/internal/registry.go

```diff
@@ -0,0 +1,96 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package olm
+
+import (
+	"context"
+	"fmt"
+
+	olmclient "github.com/operator-framework/operator-sdk/internal/olm/client"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// SDKLabels are used to identify certain operator-sdk resources.
+var SDKLabels = map[string]string{
+	"owner": "operator-sdk",
+}
+
+// RegistryResources configures creation/deletion of internal registry-related
+// resources.
+type RegistryResources struct {
+	Client  *olmclient.Client
+	Pkg     registry.PackageManifest
+	Bundles []*registry.Bundle
+}
+
+// FEAT(estroz): allow users to specify labels for registry objects.
+
+// CreateRegistryManifests creates all registry objects required to serve
+// manifests from m.manifests in namespace.
+func (m *RegistryResources) CreateRegistryManifests(ctx context.Context, namespace string) error {
+	pkgName := m.Pkg.PackageName
+	binaryKeyValues, err := createConfigMapBinaryData(m.Pkg, m.Bundles)
+	if err != nil {
+		return fmt.Errorf("error creating registry ConfigMap binary data: %w", err)
+	}
+	cm := newRegistryConfigMap(pkgName, namespace,
+		withBinaryData(binaryKeyValues),
+	)
+	volName := getRegistryVolumeName(pkgName)
+	dep := newRegistryDeployment(pkgName, namespace,
+		withRegistryGRPCContainer(pkgName),
+		withVolumeConfigMap(volName, cm.GetName()),
+		withContainerVolumeMounts(volName, []string{containerManifestsDir}),
+	)
+	service := newRegistryService(pkgName, namespace,
+		withTCPPort("grpc", registryGRPCPort),
+	)
+	if err = m.Client.DoCreate(ctx, cm, dep, service); err != nil {
+		return fmt.Errorf("error creating operator %q registry-server objects: %w", pkgName, err)
+	}
+	depKey := types.NamespacedName{
+		Name:      dep.GetName(),
+		Namespace: namespace,
+	}
+	log.Infof("Waiting for Deployment %q rollout to complete", depKey)
+	if err = m.Client.DoRolloutWait(ctx, depKey); err != nil {
+		return fmt.Errorf("error waiting for Deployment %q to roll out: %w", depKey, err)
+	}
+	return nil
+}
+
+// DeleteRegistryManifests deletes all registry objects serving manifests
+// from m.manifests in namespace.
+func (m *RegistryResources) DeleteRegistryManifests(ctx context.Context, namespace string) error {
+	pkgName := m.Pkg.PackageName
+	cm := newRegistryConfigMap(pkgName, namespace)
+	dep := newRegistryDeployment(pkgName, namespace)
+	service := newRegistryService(pkgName, namespace)
+	err := m.Client.DoDelete(ctx, dep, cm, service)
+	if err != nil {
+		return fmt.Errorf("error deleting operator %q registry-server objects: %w", pkgName, err)
+	}
+	return nil
+}
+
+// GetRegistryServiceAddr returns a Service's DNS name + port for a given
+// pkgName and namespace.
+func GetRegistryServiceAddr(pkgName, namespace string) string {
+	name := getRegistryServerName(pkgName)
```

Comment:
Why is required to export this one? I could not find other places where it is used. Actually where it has been used?

Reply:
[Here](https://github.com/operator-framework/operator-sdk/pull/1912/files#diff-59b535800ca0d926da38b8e79bc87818R209).

Reply:
So, you are implementing here a func that will be used in another PR?  
So, I think that this implementation should be in https://github.com/operator-framework/operator-sdk/pull/1912 and this PR blocked for this one.

Reply:
It _is_ in #1912, but I moved it here to make #1912 easier to review.

Reply:
I'm okay with it being exported as these will be used by future PRs.

Reply:
I am ok too after the clarifications. 
I closing as resolved this one.